Updated v2_18_2.md (#4630)
zowe/docs-site: docs/whats-new/release-notes/v2_18_2.md

## Bug fixes

Zowe Version 2.18.2 contains the bug fixes that are described in the following topics:

### Zowe Application Framework

> - Fixed an issue of the API Catalog by introduced a workaround. Before the fix, the API Catalog attempted to establish two secure Transport Layer Security (TLS) connections while the API Catalog operated under AT-TLS. Since AT-TLS already handled the secure communication, the attempt by the API Catalog was unnecessary and led to performance overheads or failures in communication. After the fix, the registration information of the API Catalog updates in the Eureka client system to indicate to other systems that AT-TLS is in use and provides secure communication. This ensures that the API Catalog and other services interacting with it recognize the existing secure environment that AT-TLS provides and avoid unnecessary security processing. ([#609](https://github.com/zowe/zlux-server-framework/pull/609))

Reviewer: ```suggestion
- Fixed an issue of the API Catalog by introducing a workaround. Before the fix, the API Catalog attempted to establish two secure Transport Layer Security (TLS) connections while the API Catalog operated under AT-TLS. Since AT-TLS already handled the secure communication, the attempt by the API Catalog was unnecessary and led to performance overheads or failures in communication. After the fix, the registration information of the API Catalog updates in the Eureka client system to indicate to other systems that AT-TLS is in use and provides secure communication. This ensures that the API Catalog and other services interacting with it recognize the existing secure environment that AT-TLS provides and avoids unnecessary security processing. ([#609](https://github.com/zowe/zlux-server-framework/pull/609))
```

Author: @anaxceron Done, thanks for the review.

### ZSS

- Fixed an issue where the HTTP client software was unable to handle `EWOULDBLOCK` errors while processing data from AT-TLS in Zowe. The fix introduced `HTTP_CLIENT_UNBLOCKED_TRY_AGAIN (19)` and `HTTP_CLIENT_SOCKET_TIMEOUT (20)` error codes into the HTTP client software to enable the HTTP client software to handle the `EWOULDBLOCK` errors with greater efficiency. ([#534](https://github.com/zowe/zowe-common-c/pull/534))

> - Fixed an issue where there was a delay in the response from the Gateway after HTTP or AT-TLS contacted the Gateway for Single Sign On(SSO). After the fix, if HTTP or AT-TLS contact the Gateway for SSO, the Gateway provides an immediate response. ([#775](https://github.com/zowe/zss/pull/775)), ([#772](https://github.com/zowe/zss/pull/772))

Reviewer: ```suggestion
- Fixed an issue where there was a delay in the response from the Gateway after HTTP or AT-TLS contacted the Gateway for Single Sign On (SSO). After the fix, if HTTP or AT-TLS contacts the Gateway for SSO, the Gateway provides an immediate response. ([#775](https://github.com/zowe/zss/pull/775)), ([#772](https://github.com/zowe/zss/pull/772))
```

Author: @anaxceron Done, thanks for the review.

> - Fixed an issue where Zowe, which was installed on the zOS operating system, stopped working because that version of zOS changed or upgraded. After the fix, Zowe continues to work uninterrupted even if the version of the underlying zOS changes or upgrades. ([#780](https://github.com/zowe/zss/pull/780))

Reviewer: ```suggestion
- Fixed an issue where Zowe, when installed on the z/OS operating system, stopped working because that version of z/OS changed or upgraded. After the fix, Zowe continues to work uninterrupted even if the version of the underlying z/OS changes or upgrades. ([#780](https://github.com/zowe/zss/pull/780))
```
